shotover-proxy/src/transforms/cassandra/peers_rewrite.rs

```diff
@@ -67,15 +78,60 @@ fn rewrite_port(message: &mut Message, new_port: u32) {
         if let CassandraOperation::Result(CassandraResult::Rows { value, metadata }) =
             &mut frame.operation
         {
-            let port_column_index = metadata
-                .col_specs
+            let columns = ["native_port", "rpc_port", "broadcast_port", "peer_port"];
+
+            let port_column_indices = columns
                 .iter()
-                .position(|col| col.name.as_str() == "native_port");
+                .map(|col| {
+                    metadata
+                        .col_specs
+                        .iter()
+                        .position(|wanted_col| &wanted_col.name.as_str() == col)
+                })
+                .flatten()
+                .collect::<Vec<_>>();
 
-            if let Some(i) = port_column_index {
+            if let MessageValue::Rows(rows) = &mut *value {
+                for i in &port_column_indices {
+                    for row in rows.iter_mut() {
+                        row[*i] = MessageValue::Integer(new_port as i64, IntSize::I32);
+                    }
+                }
+                message.invalidate_cache();
```

Comment:
Maybe just stick a single invalidate_cache call here?
![image](https://user-images.githubusercontent.com/5120858/157555328-0412aa58-482f-4520-a773-3b37703fbba3.png)
If the system.peers is behaving correctly then it should be fine to just always invalidate the cache on its responses.

As its done currently we have some level of protection to only invalidate cache when a modification occurs so at first glance it looks like a possible bug that we dont have full protection here.
So I think its better to just go the other way and always invalidate the cache.